### PR TITLE
Investigate why extension methods without type attribute not been found when defined in a different project

### PR DESCRIPTION
**Pull request by edgarfgp**
Follow up: https://github.com/dotnet/fsharp/pull/13839

### CODE REVIEW

File: tests/FSharp.Compiler.ComponentTests/Language/ExtensionMethodTests.fs [hunk 1]
```diff
@@ -114,6 +114,43 @@ namespace Consumer
...
+
+open System.Runtime.CompilerServices
+
+type WidgetBuilder<'msg, 'marker>() = class end
+
+type IMarkerOne = interface end
+
+// Commenting out [<Extension>] breaks
+//[<Extension>]
+type WidgetBuilderExtensions =
```
Comment (nojaf) [inline tests/FSharp.Compiler.ComponentTests/Language/ExtensionMethodTests.fs:132]: The problem here is when unpickling the binary of the `Producer`, there is no attribute present on the type.

In https://github.com/dotnet/fsharp/pull/13839 we are pushing the attribute to the type, and that must somehow not get pickled.

Comment (nojaf) [reply tests/FSharp.Compiler.ComponentTests/Language/ExtensionMethodTests.fs:132]: image

Comment (nojaf) [reply tests/FSharp.Compiler.ComponentTests/Language/ExtensionMethodTests.fs:132]: The signature data does not have the attribute, there lies the problem.

**Comment by edgarfgp** [comment]
Closing this in favour of https://github.com/dotnet/fsharp/pull/16368

**Pull request closed without merging**